eZ/Publish/API/Repository/Tests/SetupFactory/LegacySolr.php

```diff
@@ -104,19 +104,29 @@ protected function indexAll()
         $stmt->execute();
 
         $contentObjects = array();
+        $locations = array();
         while ( $row = $stmt->fetch( PDO::FETCH_ASSOC ) )
         {
             $contentObjects[] = $persistenceHandler->contentHandler()->load(
                 $row['id'],
                 $row['current_version']
             );
+            $locations = array_merge(
+                $locations,
+                $persistenceHandler->locationHandler()->loadLocationsByContent( $row["id"] )
+            );
         }
 
         /** @var \eZ\Publish\Core\Search\Solr\Content\Handler $contentSearchHandler */
         $contentSearchHandler = $searchHandler->contentSearchHandler();
-        $contentSearchHandler->setCommit( false );
-        $contentSearchHandler->purgeIndex();
         $contentSearchHandler->setCommit( true );
```

Comment:
Then we should probably implement empty commit request, `commit()` instead of `setCommit()`.